src/Lachain.Core/Vault/KeyGenManager.cs

```diff
@@ -255,7 +256,7 @@ private void BlockManagerOnSystemContractInvoked(object _, InvocationContext con
                 Logger.LogDebug($"Detected call of GovernanceContract.{GovernanceInterface.MethodKeygenConfirm}");
                 var keygen = GetCurrentKeyGen();
                 if (keygen is null) return;
-                var sender = keygen.GetSenderByPublicKey(context.Receipt.RecoverPublicKey());
+                var sender = keygen.GetSenderByPublicKey(context.Receipt.RecoverPublicKey(HardforkHeights.IsHardfork_8Active(context.Snapshot.Blocks.GetTotalBlockHeight() + 1)));
```

Comment:
Should it be context.Receipt.Block to check hardfork8 ?

Reply:
fixed